spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/QueryUtils.java

```diff
@@ -570,6 +583,19 @@ public static String createCountQueryFor(String originalQuery) {
 	 */
 	@Deprecated
 	public static String createCountQueryFor(String originalQuery, @Nullable String countProjection) {
+		return createCountQueryFor(originalQuery, countProjection, false);
+	}
+
+	/**
+	 * Creates a count projected query from the given original query.
+	 *
+	 * @param originalQuery must not be {@literal null}.
+	 * @param countProjection may be {@literal null}.
+	 * @param nativeQuery whether the underlying query is a native query.
+	 * @return a query String to be used a count query for pagination. Guaranteed to be not {@literal null}.
+	 * @since 2.7.8
+	 */
+	static String createCountQueryFor(String originalQuery, @Nullable String countProjection, boolean nativeQuery) {
```

Comment:
I'd prefer two separate methods `createCountQueryForNative` and `createCountQueryForJpql` plus extracting the common parts into a new method.

Reply:
My gut feeling is that we should rethink `QueryUtils` entirely and eventually eliminate the most methods (or create delegates). Therefore, I would want to refrain from introducing additional methods for the time being unless new methods heavily improve the code.